src/background/background.js

```diff
@@ -112,13 +112,17 @@ browser.runtime.onMessage.addListener(async request => {
 			return browser.tabs.sendMessage(tab.id, request);
 		}
 	} else if (request.backgroundScriptCall) {
-		return new Promise(resolve => {
+		return new Promise((resolve, reject) => {
 			const backgroundScript = getBackgroundScript('BackgroundScript');
 			// TODO change to promises
 			const deferredResponse = backgroundScript[request.fn](request.request);
-			deferredResponse.done(function (resp) {
-				resolve(resp);
-			});
+			deferredResponse
+				.done(function (resp) {
```

Comment:
Use arraow function also here.Maybe we can just use it like this? `.done(resolve)`